client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoPreparedStatement.java

```diff
@@ -986,6 +994,19 @@ private void requireNonBatchStatement()
         }
     }
 
+    private String getExecuteImmediateSql(List<String> values)
+    {
+        StringBuilder sql = new StringBuilder();
+        sql.append("EXECUTE IMMEDIATE '");
+        sql.append(this.originalSql);
```

Comment:
This needs escaping, ie. single quotes needs to be replaced with double single quotes.

Reply:
Also add tests for this scenario.

Reply:
Good catch. Fixed and added the test